src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -299,20 +299,19 @@ internal static object InternalAlloc(RuntimeType type)
         [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "RuntimeTypeHandle_InternalAlloc")]
         private static unsafe partial void InternalAlloc(MethodTable* pMT, ObjectHandleOnStack result);
 
-        internal static object InternalAllocNoChecks(MethodTable* pMT)
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal static object InternalAllocNoChecks(MethodTable* pMT, out uint bytes)
         {
-            object? result = null;
-            InternalAllocNoChecks(pMT, ObjectHandleOnStack.Create(ref result));
+            object? result = GetThreadLocalAllocInternal(pMT, out bytes);
+            if (result is null)
+                InternalAllocNoChecks(pMT, ObjectHandleOnStack.Create(ref result));
             return result!;
         }
 
         internal static object InternalAllocNoChecks(RuntimeType type)
```

Comment:
This should be deleted. Instead. the one place in delegates that calls this should be changed do `InternalAllocNoChecks(RuntimeHelpers.GetMethodTable(this))`.